File: 2696 Minimum String Length After Removing Substrings/2696.cpp
class Solution {
public:
    int minLength(string s) {
        stack<char> stk;

        for (char c: s){
            if (!stk.empty() && (stk.top() == 'A' && c == 'B' || stk.top() == 'C' && c == 'D')){
                stk.pop();
            } else {
                stk.push(c);
            }
        }

        return stk.size();
    }
};

File: 2696 Minimum String Length After Removing Substrings/2696.py
class Solution:
    def minLength(self, s: str) -> int:
        ans = []

        for c in s:
            if ans and (ans[-1] == "A" and c == "B" or ans[-1] == "C" and c == "D"):
                ans.pop()
            else:
                ans.append(c)

        return len(ans)
